volatility3/framework/automagic/pdbscan.py

```diff
@@ -376,8 +377,43 @@ def check_kernel_offset(
                     valid_kernel = (virtual_layer_name, address, res[0])
         return valid_kernel
 
+    def method_low_stub_offset(self,
+        context: interfaces.context.ContextInterface,
+        vlayer: layers.intel.Intel,
+        progress_callback: constants.ProgressCallback = None,
+    ) -> Optional[ValidKernelType]:
+        kernel_hint = 0
+        kernel_base = 0
+        physical_layer = context.layers.get('memory_layer')
+
+        # try locating kernel base via x64 Low Stub in lower 1MB starting from second page (4KB)
+        # if "Discard Low Memory" setting is disabled in BIOS, the Low Stub may be at the third/fourth or further pages
+        for offset in range(0x1000,0x100000, 0x1000):
+            if 0xffffffffffff00ff & int.from_bytes(physical_layer.read(offset, 0x8), "little") != 0x00000001000600E9:
+                continue  # not _PROCESSOR_START_BLOCK->Jmp
+            potential_kernel_hint = int.from_bytes(physical_layer.read(offset + 0x70, 0x8), "little")
+            if (0xfffff80000000003 & potential_kernel_hint) != 0xfffff80000000000:
```

Comment:
Will this work with address spaces once the 64-bit space is extended past the currently reserved bits?  Will this work on a 32-bit system?

Reply:
This method is only for 64-bit kernels, but it looks like the methods list is populated correctly so that this is method is checked the current ones are used if it fails:

https://github.com/volatilityfoundation/volatility3/blob/c10572905f3f3760594db07575534a5805a10fe3/volatility3/framework/automagic/pdbscan.py#L415

Reply:
That wasn't my issue... The question was if it's 64-bit only, why not bomb out immediately if `not isinstance(vlayer, intel.Intel32e)`?  Seems like a waste to bother scanning systems you know it can't succeed for...

Reply:
(Also, the concern about support for when the higher bits are no longer reserved still stands).